tests/unit/expiry/TestDataProvider.unitTest.ts

```diff
@@ -0,0 +1,105 @@
+
+import { MESSAGES } from "../../../src/assets/Enums";
+import { DateProvider } from "../../../src/handlers/expiry/providers/DateProvider";
+import { TestDataProvider } from "../../../src/handlers/expiry/providers/TestDataProvider";
+import { HTTPError } from "../../../src/models/HTTPError";
+
+describe("TestDataProvider", () => {
+    let testDataProvider: TestDataProvider;
+    let MockTestResultsDAO: jest.Mock;
+
+    context("for getTestHistory", () => {
+        describe("when getBySystemNumber returns error", () => {
+            it("should throw error", async () => {
+            try {
+                testDataProvider = new TestDataProvider();
+                MockTestResultsDAO = jest.fn().mockImplementation(() => {
+                return {
+                    getBySystemNumber: () => {
+                        throw new Error("some error");
+                    }
+                };
+                });
+                testDataProvider.testResultsDAO = new MockTestResultsDAO();
+                await testDataProvider.getTestHistory("123");
+            } catch (error) {
+                expect.assertions(1);
+                expect(error).toEqual(new HTTPError(500, MESSAGES.INTERNAL_SERVER_ERROR));
+            }
+            });
+        });
+        describe("when getBySystemNumber returns invalid testEndTimestamp", () => {
```

Comment:
indentation is weird, we might want to add prettier as an improvement ticket I think

Reply:
fixed indentation for all unit tests.